### DISCUSSION TITLE
Routes Typescript Codegen / OpenApi

### DISCUSSION THREAD

**Discussion by ivan-angjelkoski** [nitrojs/nitro#2718] [General]
### Describe the feature

Hey, im using Nuxt for my projects and i want to use the same API for multiple Nuxt projects, but there is a problem, i dont have type safety in my other projects and i need to create and validate the types with zod...
this problem can be avoided if i can generate the types from my project that servers the api routes, so i can import the types in my other projects.

maybe something like OpenApi specs, or Zod and Typescript types?

also if there is a way to bind these types to 'useFetch' in the other projects, that would be awesome!

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by profblackjack** [comment]
nitro already supports generating openAPI? https://nitro.unjs.io/config#openapi
